MsCorePkg/MacAddressEmulationDxe/Test/SimpleNetworkProtocolNotifyTest.c

```diff
@@ -0,0 +1,192 @@
+                         responsibility of the test author to ensure that the
+                         contents are well understood by all test cases that may
+                         consume it.
+
+  @retval  UNIT_TEST_PASSED             The Unit test has completed and the test
+                                        case was successful.
+  @retval  UNIT_TEST_ERROR_TEST_FAILED  A test case assertion has failed.
+**/
+UNIT_TEST_STATUS
+EFIAPI
+SimpleNetworkProtocolNotify_AssignsMacToOnlySameSnpAsPreviously(
+  IN UNIT_TEST_CONTEXT  Context
+  )
+{
+  // Arrange
+  MAC_EMULATION_SNP_NOTIFY_CONTEXT MacContext;
+  EFI_HANDLE *DummyHandleArr;
+  EFI_SIMPLE_NETWORK_MODE Mode1;
+  EFI_SIMPLE_NETWORK_MODE Mode2;
+  EFI_SIMPLE_NETWORK_PROTOCOL Snp1;
+  EFI_SIMPLE_NETWORK_PROTOCOL Snp2;
+
+  MacContext.Assigned = TRUE;
+  SetMem(&MacContext.EmulationAddress, NET_ETHER_ADDR_LEN, 0xEE);
+  SetMem(&MacContext.PermanentAddress, NET_ETHER_ADDR_LEN, 0xAA);
+
+  Mode1.State = EfiSimpleNetworkInitialized;
+  Mode1.IfType = NET_IFTYPE_ETHERNET;
+  Mode1.MacAddressChangeable = TRUE;
+  SetMem(&Mode1.PermanentAddress, NET_ETHER_ADDR_LEN, 0xBB);
+  SetMem(&Mode1.CurrentAddress, NET_ETHER_ADDR_LEN, 0x00);
+  Snp1.Mode = &Mode1;
+  Snp1.StationAddress = SetStationAddress;
+
+  Mode2.State = EfiSimpleNetworkInitialized;
+  Mode2.IfType = NET_IFTYPE_ETHERNET;
+  Mode2.MacAddressChangeable = TRUE;
+  SetMem(&Mode2.PermanentAddress, NET_ETHER_ADDR_LEN, 0xAA);
+  SetMem(&Mode2.CurrentAddress, NET_ETHER_ADDR_LEN, 0x00);
+  Snp2.Mode = &Mode2;
+  Snp2.StationAddress = SetStationAddress;
+
+  will_return(HandleProtocol, &Snp1);
+  will_return(HandleProtocol, EFI_SUCCESS);
+
+  will_return(HandleProtocol, &Snp2);
+  will_return(HandleProtocol, EFI_SUCCESS);
+
+  will_return_always(SnpSupportsMacEmulation, TRUE);
+
+  DummyHandleArr = AllocateZeroPool(sizeof(EFI_HANDLE*)*2);
+  EFI_HANDLE DummyHandle1 = AllocateZeroPool(sizeof(EFI_HANDLE));
+  EFI_HANDLE DummyHandle2 = AllocateZeroPool(sizeof(EFI_HANDLE));
+  DummyHandleArr[0] = DummyHandle1;
+  DummyHandleArr[1] = DummyHandle2;
+
+  will_return(LocateHandleBuffer, 2);
+  will_return(LocateHandleBuffer, DummyHandleArr);
+  will_return(LocateHandleBuffer, EFI_SUCCESS);
+
+  // Act
+  SimpleNetworkProtocolNotify(NULL, (VOID*) &MacContext);
+
+  // Assert
+  assert_memory_not_equal(&Mode1.CurrentAddress, &MacContext.EmulationAddress, NET_ETHER_ADDR_LEN);
+  assert_memory_equal(&Mode2.PermanentAddress, &MacContext.PermanentAddress, NET_ETHER_ADDR_LEN);
+  assert_memory_equal(&Mode2.CurrentAddress, &MacContext.EmulationAddress, NET_ETHER_ADDR_LEN);
+
+  FreePool(DummyHandle1);
+  FreePool(DummyHandle2);
+
+  return UNIT_TEST_PASSED;
+}
+
+UNIT_TEST_STATUS
+EFIAPI 
+SimpleNetworkProtocolNotifyTestSetup (
+  IN UNIT_TEST_CONTEXT  Context
+  )
+{
+    gBS = &mBootServices;
+    gBS->LocateHandleBuffer = LocateHandleBuffer;
+    gBS->HandleProtocol = HandleProtocol;
+    gBS->RaiseTPL = RaiseTPL;
+    gBS->RestoreTPL = RestoreTPL;
```

Comment:
I see where you put these override functions in the MacAddressEmulationDxeHostTestMain.c file and declared the prototype for HandleProtocol and LocateHandleBuffer in MacAddressEmulationDxeHostTestCommon.h.  But I don't see the prototypes anywhere for the TPL functions.

I'm sure the code compiles, is there maybe a file or something missing when pushed to this sub-module?  Maybe the Common .H file is out of date?

Reply:
Good catch yes need to add them to the header.